localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -128,41 +130,48 @@ def __init__(self, version_manager: LambdaVersionManager, event_queue_url: str):
-                # Test case (happens not every time!):
-                # tests.aws.services.cloudformation.resources.test_legacy.TestCloudFormation.test_updating_stack_with_iam_role
-                messages = sqs_client.receive_message(
+        sqs_client = get_sqs_client(
+            self.version_manager.function_version, client_config=CLIENT_CONFIG
+        )
+        function_timeout = self.version_manager.function_version.config.timeout
+        while not self._shutdown_event.is_set():
+            try:
+                response = sqs_client.receive_message(
                     QueueUrl=self.event_queue_url,
+                    # TODO: consider replacing with short polling instead of long polling to prevent keeping connections open
+                    # however, we had some serious performance issues when tried out, so those have to be investigated first
                     WaitTimeSeconds=2,
-                    # TODO: MAYBE: increase number of messages if single thread schedules invocations
-                    MaxNumberOfMessages=1,
+                    # Related: SQS event source mapping batches up to 10 messages:
+                    # https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html
+                    MaxNumberOfMessages=10,
                     VisibilityTimeout=function_timeout + 60,
                 )
-                if not messages.get("Messages"):
+                if not response.get("Messages"):
                     continue
-                message = messages["Messages"][0]
+                LOG.debug("[%s] Got %d messages", self.event_queue_url, len(response["Messages"]))
+                # Guard against shutdown event arriving while polling SQS for messages
+                if not self._shutdown_event.is_set():
+                    for message in response["Messages"]:
+                        # NOTE: queueing within the thread pool executor could lead to double executions
+                        #  due to the visibility timeout
+                        self.invoker_pool.submit(self.handle_message, message)
 
-                # NOTE: queueing within the thread pool executor could lead to double executions
-                #  due to the visibility timeout
-                self.invoker_pool.submit(self.handle_message, message)
-        except Exception as e:
-            # TODO gateway shuts down before shutdown event even is set, so this log message might be sent regardless
-            if isinstance(e, ConnectionRefusedError) and self._shutdown_event.is_set():
-                return
-            # TODO: investigate what causes ReadTimeoutError (fixed with increasing read timeout?)
-            LOG.error(
-                "Error while polling lambda events for function %s: %s",
-                self.version_manager.function_version.qualified_arn,
-                e,
-                exc_info=LOG.isEnabledFor(logging.DEBUG),
-            )
+            except Exception as e:
```

Comment:
the 2nd key change is to catch errors in the poller and retry instead of terminating the poller upon the first error